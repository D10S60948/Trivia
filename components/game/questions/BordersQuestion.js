import React, { Component } from 'react'
import { StyleSheet, View, Text } from 'react-native'

export default class BordersQuestion extends Component {

    alpha3codeToName(code) {
        let countries = this.props.countries
        for(let i=0; i < countries.length; i++) {
            if(code == countries[i].alpha3Code) {
                return countries[i].name
            }
        }
        return ''
    }

    render() {
        let randomIndex = Math.floor(Math.random()*this.props.borders.length)
        return (
            this.props.borders.length == 0) ?
            (
                <View style={styles.container}>
                    <View style={{flexWrap: 'wrap', flexDirection: 'column', paddingHorizontal: 20}}>
                        <Text style={{ fontFamily: 'Mplus-Bold' }}>{ this.props.name }</Text>
                        <Text style={{ fontFamily: 'Mplus' }}> does not share borders with any country</Text>
                    </View>
                </View>
            )
            : 
            (
                <View style={styles.container}>
                    <View style={{flexWrap: 'wrap', flexDirection: 'row', paddingHorizontal: 20}}>
                        <Text style={{ fontFamily: 'Mplus-Bold' }}>{ this.props.name }</Text>
                        <Text style={{ fontFamily: 'Mplus' }}> and </Text>
                        <Text style={{ fontFamily: 'Mplus-Bold' }}>{this.alpha3codeToName(this.props.borders[randomIndex])}</Text>
                        <Text style={{ fontFamily: 'Mplus' }}> share border</Text>
                    </View>
                </View>
            )
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column', 
  }
});
