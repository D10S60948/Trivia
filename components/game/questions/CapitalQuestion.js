import React, { Component } from 'react'
import { StyleSheet, View, Text } from 'react-native'

export default class CapitalQuestion extends Component {

    getCountryName() {
        if(this.props.countryName == '') {
            return 'No capital'
        }
        return this.props.countryName
    }

    render() {
        return (
            <View style={{...styles.container, flexDirection: 'row', paddingHorizontal: 20}}>
                <Text style={{ fontFamily: 'Mplus' }}>What is the capital of </Text>
                <Text style={{ fontFamily: 'Mplus-Bold' }}>{this.getCountryName()}</Text>
            </View>
        )
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column', 
  }
});
