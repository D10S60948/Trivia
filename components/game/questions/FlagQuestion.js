import React, { Component } from 'react';
import { StyleSheet, View, Text } from 'react-native';

export default class FlagQuestion extends Component {

    render() {
        return (
            <View style={styles.container}>
                <Text style={{ fontFamily: 'Mplus' }}>To which country the following flag belongs to</Text>
                <Text style={{ fontSize: 70 }}>{ this.props.flag }</Text>
            </View>
        )
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column', 
  }
});
