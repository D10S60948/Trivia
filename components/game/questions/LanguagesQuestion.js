import React, { Component } from 'react'
import { StyleSheet, View, Text } from 'react-native'

export default class LanguagesQuestion extends Component {
    render() {
        return (
            <View style={styles.container}>
                <View style={{flexWrap: 'wrap', paddingHorizontal: 20}}>
                    <Text style={{ fontFamily: 'Mplus' }}>The spoken languages in { this.props.name } are: </Text>
                </View>
                <View style={{flexDirection: 'row'}}>
                    { this.props.languages.map((lang, index) => 
                    <Text key={index} style={{ fontFamily: 'Mplus-Bold'}}>
                    {
                        index > 0 ?  
                            ', ' + this.props.languages[index].name  :  
                            this.props.languages[index].name
                    }
                    </Text>
                    )}
                </View>
            </View>
        )
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column', 
  }
});
