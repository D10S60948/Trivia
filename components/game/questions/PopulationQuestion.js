import React, { Component } from 'react';
import { StyleSheet, View, Text } from 'react-native';

export default class PopulationQuestion extends Component {

    render() {
        return (
            <View style={styles.container}>
                <View style={{flexWrap: 'wrap', paddingHorizontal: 20}}>
                    <Text style={{ fontFamily: 'Mplus' }}>The population of { this.props.name } is </Text>
                </View>
            </View>
        )
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column', 
  }
});
