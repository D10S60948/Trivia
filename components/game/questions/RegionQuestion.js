import React, { Component } from 'react';
import { StyleSheet, View, Text } from 'react-native';

export default class RegionQuestion extends Component {

    render() {
        return (
            <View style={styles.container}>
                <View style={{flexWrap: 'wrap', flexDirection: 'row', paddingHorizontal: 20}}>
                    <Text style={{ fontFamily: 'Mplus-Bold' }}>{ this.props.name }</Text>
                    <Text style={{ fontFamily: 'Mplus' }}> is in </Text>
                    <Text style={{ fontFamily: 'Mplus-Bold' }}>{ this.props.region }</Text>
                </View>
            </View>
        )
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column', 
  }
});
