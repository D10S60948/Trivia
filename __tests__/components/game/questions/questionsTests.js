import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import FlagQuestion from '../../../../components/game/questions/FlagQuestion';
import RegionQuestion from '../../../../components/game/questions/RegionQuestion';
import PopulationQuestion from '../../../../components/game/questions/PopulationQuestion';
import LanguagesQuestion from '../../../../components/game/questions/LanguagesQuestion';
import CapitalQuestion from '../../../../components/game/questions/CapitalQuestion';
import BordersQuestion from '../../../../components/game/questions/BordersQuestion';

test('"FlagQuestion" component renders correctly', () => {
  const snapshot = renderer.create(<FlagQuestion flag={'emoji'} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});

test('"RegionQuestion" component renders correctly', () => {
  const snapshot = renderer.create(<RegionQuestion region={'Asia'} name={'Israel'} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});

test('"PopulationQuestion" component renders correctly', () => {
  const snapshot = renderer.create(<PopulationQuestion name={'Israel'} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});

test('"LanguagesQuestion" component renders correctly', () => {
  const snapshot = renderer.create(<LanguagesQuestion languages={['English', 'Hebrew']} name={'Israel'} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});

test('"CapitalQuestion" component renders correctly', () => {
  const snapshot = renderer.create(<CapitalQuestion countryName={'Israel'} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});

test('"BordersQuestion" component renders correctly', () => {
  const snapshot = renderer.create(<BordersQuestion countries={[]} borders={['EGY', 'SYR', 'JOR', 'LBN']} name={'Israel'} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});

