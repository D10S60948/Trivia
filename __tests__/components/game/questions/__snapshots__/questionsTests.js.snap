// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"BordersQuestion" component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "paddingHorizontal": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Mplus-Bold",
        }
      }
    >
      Israel
    </Text>
    <Text
      style={
        Object {
          "fontFamily": "Mplus",
        }
      }
    >
       and 
    </Text>
    <Text
      style={
        Object {
          "fontFamily": "Mplus-Bold",
        }
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "fontFamily": "Mplus",
        }
      }
    >
       share border
    </Text>
  </View>
</View>
`;

exports[`"CapitalQuestion" component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontFamily": "Mplus",
      }
    }
  >
    What is the capital of 
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "Mplus-Bold",
      }
    }
  >
    Israel
  </Text>
</View>
`;

exports[`"FlagQuestion" component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontFamily": "Mplus",
      }
    }
  >
    To which country the following flag belongs to
  </Text>
  <Text
    style={
      Object {
        "fontSize": 70,
      }
    }
  >
    emoji
  </Text>
</View>
`;

exports[`"LanguagesQuestion" component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flexWrap": "wrap",
        "paddingHorizontal": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Mplus",
        }
      }
    >
      The spoken languages in 
      Israel
       are: 
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Mplus-Bold",
        }
      }
    />
    <Text
      style={
        Object {
          "fontFamily": "Mplus-Bold",
        }
      }
    >
      , undefined
    </Text>
  </View>
</View>
`;

exports[`"PopulationQuestion" component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flexWrap": "wrap",
        "paddingHorizontal": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Mplus",
        }
      }
    >
      The population of 
      Israel
       is 
    </Text>
  </View>
</View>
`;

exports[`"RegionQuestion" component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "paddingHorizontal": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Mplus-Bold",
        }
      }
    >
      Israel
    </Text>
    <Text
      style={
        Object {
          "fontFamily": "Mplus",
        }
      }
    >
       is in 
    </Text>
    <Text
      style={
        Object {
          "fontFamily": "Mplus-Bold",
        }
      }
    >
      Asia
    </Text>
  </View>
</View>
`;
